File: src/main/java/me/Oxygen/modules/combat/Reach.java
package me.Oxygen.modules.combat;

import me.Oxygen.modules.Category;
import me.Oxygen.modules.Module;
import me.Oxygen.modules.ModuleRegister;
import me.Oxygen.value.Value;

@ModuleRegister(name = "Reach", category = Category.COMBAT)
public class Reach extends Module {
	
	public static Value<Double> range = new Value<Double>("Reach_Range", 4.5d, 3d, 10.0d, 0.1d);

}

File: src/main/java/cn/rainbow/oxygen/module/modules/combat/Reach.kt
package cn.rainbow.oxygen.module.modules.combat

import cn.rainbow.oxygen.module.setting.NumberValue
import cn.rainbow.oxygen.Oxygen
import cn.rainbow.oxygen.module.Category
import cn.rainbow.oxygen.module.Module

object Reach : Module("Reach", Category.Combat) {

    var numberValue = NumberValue("Reach", 3.0, 3.0, 6.0, 0.1)

    @JvmStatic
    fun getReach(): Float {
        return if (Oxygen.INSTANCE.moduleManager.getModule(this.javaClass)!!.enabled) numberValue.currentValue.toFloat() else 3.0f
    }
}